blackbox/handlers/databases/zip.py

```diff
@@ -0,0 +1,41 @@
+import datetime
+from collections import deque
+from pathlib import Path
+from zipfile import ZIP_DEFLATED
+from zipfile import ZipFile
+
+from blackbox.exceptions import ImproperlyConfigured
+from blackbox.handlers.databases._base import BlackboxDatabase
+
+
+class Zip(BlackboxDatabase):
+    """A Database handler that will zip a local folder."""
+
+    required_fields = ("path",)
+
+    def backup(self) -> Path:
+        date = datetime.date.today().strftime("%d_%m_%Y")
+
+        path = self.config["path"]
+        compression_level = self.config.get("compression_level", 5)
+
+        if compression_level < 0 or compression_level > 9:
+            raise ImproperlyConfigured(
+                f"Invalid compression level. Must be an integer between 0 and 9, got {compression_level}."
+            )
```

Comment:
I think this is too late to raise this error. At this point, we might already have backed things up to several databases.

Maybe we can do this in the `__init__` for this model, instead? That way it should fail instantly, before taking any actions.